lucille-core/src/main/java/com/kmwllc/lucille/connector/jdbc/DatabaseConnector.java

```diff
@@ -118,15 +118,17 @@ private int getIdColumnIndex(String[] columns) throws ConnectorException {
 
   @Override
   public void execute(Publisher publisher) throws ConnectorException {
-
+    ResultSet rs = null;
+    ArrayList<ResultSet> otherResults = null;
+    Connection connection = null;
     try {
-      // connect to the database.
-      Connection connection = createConnection();
+      connection = createConnection();
       // run the pre-sql (if specified)
       runSql(connection, preSql);
       // TODO: make sure we cleanup result set/statement/connections properly.
-      ResultSet rs;
+
       log.info("Running primary sql");
+
       try {
         Statement state = connection.createStatement(ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY);

```

Comment:
statement is never closed

Reply:
Closed in new commit, didn't do it in a try-with-resources because the statement is being used outside of the scope of the try-catch